### PR TITLE
[patch] Added LOGS and Updated Trace for MHUB Metadata Connection 

### PR DESCRIPTION
**Pull request by baoquocphan**
- Added LOG points to for Metaconnection connections. 
- Updated TRACE for Metadata Connection
- Added retry code when creating Metadata Connection failed.

### CODE REVIEW

File: server_proxy/src/pxmhub.c [hunk 1]
```diff
@@ -3717,12 +3727,18 @@ static int mhubReceiveMetadata(ism_transport_t * transport, char * inbuf, int bu
...
+								transport->index, transport->name, transport->server_addr, transport->serverport,
+											(mhub->trybroker>0)?mhub->brokers[mhub->trybroker-1]:mhub->brokers[0]);
         transport->close(transport, ISMRC_BadClientData, 0, "MessageHub metadata incomplete");
 
         ism_mhub_lock(mhub);
-        if(!g_shuttingDown && mhub->enabled==1)
-        		ism_common_setTimerOnce(ISM_TIMER_LOW, (ism_attime_t)mhubRetryConnect, mhub, retryDelay(mhub->retry++));
-        ism_mhub_unlock(mhub);
+		if(!g_shuttingDown && mhub->enabled==1){
+			ism_mhub_unlock(mhub);
```
Comment (IanBoden) [inline server_proxy/src/pxmhub.c:3737]: I'm not sure about this, don't we want to hold the lock until after we have set the timer?

Comment (IanBoden) [reply server_proxy/src/pxmhub.c:3737]: think the version added in line 3800 also needs to change

Comment (baoquocphan) [reply server_proxy/src/pxmhub.c]: Updated the code based on the comments

**Review by jonquark — APPROVED** [review]
lgtm

**Pull request merged**